src/Transpiler.ts

```diff
@@ -1203,10 +1203,21 @@ export abstract class LuaTranspiler {
 
         }
 
-        if (tsHelper.isArrayType(ownerType, this.checker)) {
+        // if ownerType is a array, use only supported functions
+        if (tsHelper.isArrayType(ownerType, this.checker, true)) {
             return this.transpileArrayCallExpression(node);
         }
 
+        // if ownerType inherits from an array, use array calls where appropriate
+        if (tsHelper.isArrayType(ownerType, this.checker)) {
+            const transpiledExpression = this.transpileArrayCallExpression(node, true);
+            if (transpiledExpression !== "") {
+                return transpiledExpression;
+            } else {
+                // non-array method
```

Comment:
No need for an empty clause, just remove the `else`

Reply:
Also like I said below it's probably nicer to remove the need for this if statement using `tsHelper.isDefaultArrayCallExpression()`